src/service/extensions-impl.js

```diff
@@ -410,6 +410,12 @@ export class Extensions {
         // becomes necessary. This will require refactoring of extension
         // loader that can be resolved via the parent ampdoc.
 
+        // TODO(choumx): Similar to `adoptServicesForEmbed`, should there be
```

Comment:
I don't think we can have a mapping like this. Maybe instead the extension itself can somehow mark itself as "adoptable to embed"? Or an extension could register an adopter?